core/src/main/java/com/linecorp/armeria/server/DependencyInjectorEntry.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.server;
+
+/**
+ * The {@link DependencyInjector} and {@code shutdownOnStop} that tells whether
+ * the {@link DependencyInjector} is shut down when the {@link Server} stops.
+ */
+public final class DependencyInjectorEntry {
```

Comment:
Is it possible to move `shutdownOnStop` to `DependencyInjector`?

For example:
```java
public interface DependencyInjector extends SafeCloseable {

    ...

    void shutdownOnStop(boolean shutdownOnStop);

    boolean shutdownOnStop();
}
```

Reply:
Otherwise, we can add a generalized helper class to support shutdown.
```java
public interface ShutdownSupport<T> extends SafeCloseable {

    static of(T object, boolean shutdownOnStop) { ... }

    static of(T object) { ... }

    void shutdownOn(Server);

    ServerListener toServerListener();
}

public ServerBuilder dependencyInjector(DependencyInjector dependencyInjector, boolean shutdownOnStop) {
    requireNonNull(dependencyInjector, "dependencyInjector");
    // Use ShutdownSupport instead of DependencyInjectorEntry
    dependencyInjectors.add(new ShutdownSupport.of(dependencyInjector, shutdownOnStop));
    return this;
}
```

Reply:
The latter sounds better to me.

Reply:
I created another [PR](https://github.com/line/armeria/pull/4280) based on the idea. Thanks!